witness/src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -44,51 +47,60 @@ expandGTxs ::
        ( Default (ChgAccum ctx)
        , HasLens ctx RestrictCtx
        , HasLens ctx (ChgAccumCtx ctx)
+       , HasCoreConfig
        )
     => [GTxWitnessed]
     -> ERoComp Exceptions Ids Values ctx [StateTx Ids Proofs Values]
-expandGTxs txs = expandUnionRawTxs getByGTx txs
+expandGTxs = expandUnionRawTxs getByGTx
 
--- | Expand list of global txs.
+-- | Expand one global tx.
 expandGTx ::
        ( Default (ChgAccum ctx)
        , HasLens ctx RestrictCtx
        , HasLens ctx (ChgAccumCtx ctx)
+       , HasCoreConfig
        )
     => GTxWitnessed
     -> ERoComp Exceptions Ids Values ctx (StateTx Ids Proofs Values)
-expandGTx txs =
-    expandUnionRawTxs getByGTx [txs] >>= \case
+expandGTx tx =
+    expandUnionRawTxs getByGTx [tx] >>= \case
         [expanded] -> return expanded
         _          -> error "expandGTx: did not expand 1 raw tx into 1 sd tx"
 
-getByGTx ::
-       GTxWitnessed
+getByGTx
+    :: HasCoreConfig
+    => GTxWitnessed
     -> ( StateTxType
        , Proofs
-       , SeqExpanders Exceptions Ids Proofs Values ctx GTxWitnessed)
-getByGTx t = case t of
-    GMoneyTxWitnessed tx ->
-        let (a, b) = toProofBalanceTx tx
-        in  (a, b, seqExpandersGTx t)
-
-    GPublicationTxWitnessed ptx ->
-        let (a, b) = toProofPublicationTx ptx
-        in  (a, b, seqExpandersGTx t)
-
-toProofPublicationTx :: PublicationTxWitnessed -> (StateTxType, Proofs)
-toProofPublicationTx (PublicationTxWitnessed tx (PublicationTxWitness {..})) =
+       , SeqExpanders Exceptions Ids Proofs Values ctx GTxWitnessed
+       )
+getByGTx t =
+    let size = BS.length $ serialise t
+        fee  = calculateFee feeCoefficients size
```

Comment:
By design, the fee for publications is calculated differently than fee for money transactions - it should depend on the size of the private block, encoded in the private header. We can leave it like this for alpha, but you should bear this in mind.